backend/packages/Upgrade/src/api/services/ExperimentService.ts

```diff
@@ -405,6 +406,20 @@ export class ExperimentService {
     }
   }
 
+  private hasConditionCodeDefault(conditions: ExperimentCondition[]): any {
+    // Check for conditionCode is 'default' then return error:
+    const hasDefaultConditionCode = conditions.filter(
+      condition => condition.conditionCode.toUpperCase() === 'DEFAULT'
+    );
+    if (!!hasDefaultConditionCode.length) {
```

Comment:
Don't we need just a single ! instead of !!?

Reply:
`!` returns true if the object is null, undefined or an empty string and false otherwise. If you negate it again using `!!` you get true for values that exist and false for the ones that do not

Reply:
I feel this is fine as when we have a condition filtered with 'default' name, then `!!` this would allow it to enter if condition and throw badrequesterror

Reply:
@ppratikcr7 Can you check this logic again by keeping only `hasDefaultConditionCode.length` only because filter function will return back the array always and if you have anything in the array then it will throw error. So it should work by removing !!.

Reply:
okay let me check

Reply:
Code changes done